### PR TITLE
feat: osco modernize

### PR DESCRIPTION
**Pull request by degenaro**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

This PR aligns the osco-to-oscal transformer with the tanium-to-oscal transformer.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar

### CODE REVIEW

File: trestle/transforms/implementations/osco.py [hunk 1]
```diff
@@ -87,28 +127,434 @@ def _ingest_yaml(self, blob: str) -> Results:
...
+                break
+        return value
+
+    def _get_target_type(self, root: Element) -> str:
+        """Extract target_type from the XML."""
+        value = None
+        benchmark_href = self._get_benchmark_href(root)
+        if benchmark_href is not None:
+            value = benchmark_href.split('-')[1]
+        return value
```
Comment (fsuits) [inline trestle/transforms/implementations/osco.py:211]: Should this be checked to have a - in it, or try block?

File: trestle/transforms/implementations/osco.py [hunk 2]
```diff
@@ -87,28 +127,434 @@ def _ingest_yaml(self, blob: str) -> Results:
...
+    def _get_fact(self, lev1: Element, kw: str) -> str:
+        """Extract fact from the XML."""
+        value = None
+        for lev2 in lev1:
+            tag = _remove_namespace(lev2.tag)
+            if tag == 'fact':
+                name = lev2.get('name')
+                if name == kw:
+                    value = lev2.text
+                    break
```
Comment (fsuits) [inline trestle/transforms/implementations/osco.py:240]: is name sure to be present?

**Review by fsuits — APPROVED** [review]
LGTM - just some minor comments on queries for things that may not be present.  Try blocks might be needed to give meaningful error msg if there is a chance items are not as expected.

File: trestle/transforms/implementations/osco.py (hunk 2 above)
Comment (degenaro) [reply trestle/transforms/implementations/osco.py:240]: I think this is OK. If name is not found, None comes back. The `kw` should never be None else that's a code bug not a data error.

**Review by butler54 — APPROVED** [review]
LGTM

**Pull request merged**